src/store/task/reducers/taskReducers.js

```diff
@@ -47,6 +48,21 @@ export const taskReducer = (state = initialTasksState, action) => {
       };
     }
 
+    case EDIT_TASK: {
+      const updatedTasks = state.tasks.map((task) => {
+        if (task.taskId === action.payload.taskId) {
+          return {
+            ...task,
+            taskTitle: action.payload.taskTitle,
+          };
+        }
+
+        return { ...task };
```

Comment:
Is there any reason to destructuring task here ?

Reply:
```
const updatedTasks = state.tasks.map((task) => {
        if (task.taskId === action.payload) {
          return {
            ...task,
            isTaskDone: true,
            taskDoneAt: new Date(),
          };
        }

        return { ...task };
      });
```

@azo-wtag vai, if it's the desired task then we are updating the `isTaskDone` and `taskDoneAt` property. Otherwise there's no update at all. And as task is stored inside an array which is inside an object, we are destructuring to keep the rest as it is.

Reply:
do we need to destructure task if no update is required ? @nsi-wtag vai

Reply:
@azo-wtag vai, shouldn't we store a fresh new copy each time! `task` is an object with multiple properties. Is there any alternative?

Reply:
Out goal is to make reducer a pure function. To achieve that if we update an object then we need to destructure it as the value outside the reducer could be updated in this case. But if any task remains intact then destructuring & creating a new object could be redundant. May be we can discuss it offline. You can have a look a [redux-immutability pattern](https://redux.js.org/usage/structuring-reducers/immutable-update-patterns) for reference. Thank you, @nsi-wtag vai